yellowbrick/features/rankd.py

```diff
@@ -240,27 +275,173 @@ def rank(self, X, algorithm=None):
                 "'{}' is unrecognized ranking method".format(algorithm)
             )
 
+        # Extract matrix from dataframe if necessary
+        if is_dataframe(X):
```

Comment:
Also, you mentioned sorting in the original PR; if the user specifies an order in `features`, it should be maintained; but if any sorting occurs (probably an option to pass to `__init__`, then I think it should also be done here! What do you think?